lib/cretonne/src/isa/intel/abi.rs

```diff
@@ -168,6 +169,31 @@ pub fn callee_saved_registers(flags: &shared_settings::Flags) -> &'static [RU] {
     }
 }
 
+fn callee_saved_registers_used(
+    flags: &shared_settings::Flags,
+    func: &ir::Function,
+) -> AllocatableSet {
+    let mut all_callee_saved = AllocatableSet::empty();
+    for reg in callee_saved_registers(flags) {
+        all_callee_saved.free(GPR, *reg as RegUnit);
+    }
+
+    let mut used = AllocatableSet::empty();
+    for value_loc in func.locations.values() {
+        match value_loc {
+            &ValueLoc::Reg(ru) => {
+                if !used.is_avail(GPR, ru) {
+                    used.free(GPR, ru);
```

Comment:
If we can think of a good name, it'd be nice to add a function to `AllocatableSet` aka `RegisterSet` that adds a register to the set without asserting that it's not already set.